engine/components/IgePathComponent.ts

```diff
@@ -708,7 +708,7 @@ export class IgePathComponent extends IgeComponent {
 						// We must translate the entity at a minimum once per point to ensure it's coords are correct if a new path starts
 						this._entity.translateToPoint(newPoint);
 
-						this.emit("pointComplete", this._entity, pointArr[p].x, pointArr[p].y, pointNext.x, pointNext.y, p, p + 1, effectiveTime);
+						this.emit('pointComplete', [this, pointArr[p].x, pointArr[p].y, pointNext.x, pointNext.y, p, p + 1, effectiveTime, pointArr[p].z, pointNext.z]);
```

Comment:
This update might be incorrect. Looks like you're wrapping the emitted data into an array, but the `emit()` function allows multiple arguments to be passed. When received with the `on("pointComplete", () => {})` you can have it in array format by using a rest operator e.g `on("pointComplete", (...args) => {})`. Otherwise, each arg can be named instead. I'll be updating IgeEventingClass soon to allow type-safe events by name as well so when listening for "pointComplete" your callback function will have typed arguments.

Finally, the `this` argument vs `this._entity`... I can guess why you might want this, but I'd like to understand the reasoning a bit more, if you could show me why you need the path component directly? It makes sense to provide more access than less, but it's a breaking change so just checking!

Reply:
Or we can pass the objects directly e.g.

```
this.emit(
	"pointComplete",
	this._entity,
	pointArr[p],
	pointNext,
	p,
	p + 1,
	effectiveTime
);
```

Reply:
Also, there are other events currently being raised like "pathRecalculated". I'm gonna update those to the form above as well passing the object directly `pointArr[p]` rather than the individual component x,y,z parts

Reply:
The other thing about `this` vs `this._entity` is that all the events should be uniform. Gonna leave them as `this._entity` for the moment until I can have a chat with you :)